src/Core/Entity/Role.php

```diff
@@ -37,6 +39,17 @@ class Role
     #[ORM\ManyToMany(targetEntity: User::class, mappedBy: 'roles', fetch: 'EXTRA_LAZY')]
     private Collection $users;
 
+    /**
+     * @var Collection<Permission>
+     */
+    #[ORM\ManyToMany(targetEntity: Permission::class, mappedBy: 'roles')]
+    private Collection $permissions;
```

Comment:
So if `Permission` isn't an entity anymore, this will probably just become a `simple_array`.

Some pointers for `simple_array` because it's got some pitfalls with empty arrays:
```
#[ORM\Column(type: 'simple_array', nullable: true)] // you need nullable because doctrine converts [] to null
private ?array $permissions = [];
```

And then a getter/setter for simple_array is a bit funky. Note that the getter and setter should not have the null, like so:
```php
public function getPermissions(): array
{
    return $this->permissions ?? [];
}

public function setPermissions(array $permissions): void
{
    $this->permissions = $permissions; // with setter it's fine because of the doctrine conversions from [] to null
}
```